crates/provider-sdk/src/lib.rs

```diff
@@ -203,9 +204,45 @@ impl LinkConfig
     }
 }
 
+/// Configuration object is made available when a provider is started, to assist in init
+///
+/// This trait exists to both obscure the underlying implementation and control what information
+/// is made available
+pub trait ProviderInitConfig: Send + Sync {
+    /// Get host-configured provider ID.
+    ///
+    /// This value may not be knowable to the provider at build time but must be known by runtime.
+    fn get_provider_id(&self) -> &str;
+
+    /// Retrieve the static configuration for the provider
+    ///
+    /// This normally consists of named configuration that were set for the provider,
+    /// merged, and received from the host *before* the provider has started initialization.
+    fn get_static_config(&self) -> &HashMap<String, String>;
```

Comment:
Do you think it's a bad name calling it "static" config if we might support updating it in the future? Or is that acceptable? I know this is what I've been calling it, but now is a good time for a gut check 😄  cc @thomastaylor312

Reply:
I can definitely rename this `config` but I'd note that this `trait` in particular is named very specifically for *init* -- I don't expect it to be called again. 

Definitely happy to change this to just `get_config()` though, so will do that!

Reply:
Perfect 😄 We do eventually want to support updating this config in the future, so perhaps we can add another trait method here like "update" where providers can opt into updates to config

Reply:
Hey sorry a bit late getting back to this, but can we add that when the functionality exists? This was really more about `init()` only

Reply:
Filing this so we don't forget it:

https://github.com/wasmCloud/wasmCloud/issues/1747